dm/loader/lightning.go

```diff
@@ -231,10 +233,16 @@ func (l *LightningLoader) restore(ctx context.Context) error {
 			return err
 		}
 		cfg.Routes = l.cfg.RouteRules
-		cfg.Checkpoint.Driver = lcfg.CheckpointDriverFile
-		cpPath := filepath.Join(l.cfg.LoaderConfig.Dir, lightningCheckpointFileName)
-		cfg.Checkpoint.DSN = cpPath
-		cfg.Checkpoint.KeepAfterSuccess = lcfg.CheckpointOrigin
+
+		// TODO lightning checkpoint will support s3 in other pr
+		if isS3, _ := exstorage.IsS3Path(l.cfg.LoaderConfig.Dir); isS3 {
```

Comment:
sees we can change func `IsS3Path` only return bool? when url parse faild, the input string must not be the s3-path format

Reply:
yes, it seems to be better.